packages/backend-common/src/reading/BitbucketServerUrlReader.ts

```diff
@@ -83,22 +84,28 @@ export class BitbucketServerUrlReader implements UrlReader {
 
     let response: Response;
     try {
-      response = await fetch(bitbucketUrl.toString(), {
-        headers: {
-          ...requestOptions.headers,
-          ...(etag && { 'If-None-Match': etag }),
-          ...(lastModifiedAfter && {
-            'If-Modified-Since': lastModifiedAfter.toUTCString(),
-          }),
-        },
-        // TODO(freben): The signal cast is there because pre-3.x versions of
-        // node-fetch have a very slightly deviating AbortSignal type signature.
-        // The difference does not affect us in practice however. The cast can be
-        // removed after we support ESM for CLI dependencies and migrate to
-        // version 3 of node-fetch.
-        // https://github.com/backstage/backstage/issues/8242
-        ...(signal && { signal: signal as any }),
-      });
+      response = await pRetry(async () => {
```

Comment:
Just thinking out loud; should this be moved to some common package so that other plugins could also utilize fetch with retries? I believe there was similar problem with the github provider as well?

Reply:
I would like to test this approach with some limited number of integrations. If it will be successful we could transform it into general method.